rollouts/pkg/packagediscovery/packagediscovery.go

```diff
@@ -0,0 +1,104 @@
+package packagediscovery
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path/filepath"
+
+	gitopsv1alpha1 "github.com/GoogleContainerTools/kpt/rollouts/api/v1alpha1"
+	"github.com/google/go-github/v48/github"
+	"golang.org/x/oauth2"
+	coreapi "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PackageDiscovery struct {
+	config    gitopsv1alpha1.PackagesConfig
+	client    client.Client
+	namespace string
+}
+
+type DiscoveredPackage struct {
```

Comment:
Note to self: I am realizing we will need some consistent way of discovering `package id + version`.